### PR TITLE
CircleCI: enable OSX jobs

### PR DESCRIPTION
**Pull request by yf225**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -384,7 +384,7 @@ caffe2_macos_build_defaults: &caffe2_macos_build_defaults
 
           export IN_CIRCLECI=1
 
-          brew install moreutils
+          brew install moreutils --without-parallel
```
Comment (ezyang) [inline .circleci/config.yml]: Why? Can you document what happened without it?

File: .jenkins/pytorch/macos-build.sh [hunk 2]
```diff
@@ -14,6 +14,7 @@ if [ ! -d "${PYTORCH_ENV_DIR}/miniconda3" ]; then
 fi
 export PATH="${PYTORCH_ENV_DIR}/miniconda3/bin:$PATH"
 source ${PYTORCH_ENV_DIR}/miniconda3/bin/activate
+conda install python=3.6
```
Comment (ezyang) [inline .jenkins/pytorch/macos-build.sh]: This looks pretty strange. Are you... downgrading to python 3.6? When can we upgrade to Python 3.7? What exactly goes wrong without it?

File: .jenkins/pytorch/macos-test.sh [hunk 3]
```diff
@@ -15,6 +15,7 @@ if [ ! -d "${PYTORCH_ENV_DIR}/miniconda3" ]; then
 fi
 export PATH="${PYTORCH_ENV_DIR}/miniconda3/bin:$PATH"
 source ${PYTORCH_ENV_DIR}/miniconda3/bin/activate
+conda install python=3.6
```
Comment (ezyang) [inline .jenkins/pytorch/macos-test.sh]: Why don't we run the build and test all in one go? There's no GPU consideration here. Would save a lot of time reinstalling stuff again second time around too.

**Review by ezyang — CHANGES_REQUESTED** [review]
We're not in a rush. I want to know more about some of the changes.

File: .jenkins/pytorch/macos-test.sh (hunk 3 above)
Comment (yf225) [reply .jenkins/pytorch/macos-test.sh]: I am thinking about implementing the parallel testing solution soon for OSX tests (along with other Linux tests that take a long time). Separating them would be the first step on that path, although we can do them in one go for now and change it later.

File: .jenkins/pytorch/macos-build.sh (hunk 2 above)
Comment (yf225) [reply .jenkins/pytorch/macos-build.sh]: I just tested with Python 3.7 and it seems to work pretty well :)

File: .jenkins/pytorch/macos-test.sh [hunk 4]
```diff
@@ -15,7 +15,7 @@ if [ ! -d "${PYTORCH_ENV_DIR}/miniconda3" ]; then
 fi
 export PATH="${PYTORCH_ENV_DIR}/miniconda3/bin:$PATH"
 source ${PYTORCH_ENV_DIR}/miniconda3/bin/activate
-conda install -y mkl mkl-include numpy pyyaml setuptools cmake cffi ninja future six
+conda install -y mkl mkl-include numpy pyyaml setuptools cmake cffi ninja six
```
Comment (yf225) [inline .jenkins/pytorch/macos-test.sh]: `future` doesn't have Python 3.7 support, but it seems that we don't actually need it.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**